pkg/divert/driver.go

```diff
@@ -36,19 +37,25 @@ type Driver interface {
 	UpdateVirtualService(vs *istioNetworkingV1beta1.VirtualService)
 }
 
-func New(m *model.Manifest, c kubernetes.Interface) (Driver, error) {
+func New(divert *model.DivertDeploy, name, namespace string, c kubernetes.Interface) (Driver, error) {
```

Comment:
This doesn't depend from the manifest anymore, just from the minimum information needed for it to work